src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1628,7 +1628,8 @@ def create_vm(vm_name, resource_group_name, image=None, size='Standard_DS1_v2',
             master_template.add_resource(build_public_ip_resource(public_ip_address, location,
                                                                   tags,
                                                                   public_ip_address_allocation,
-                                                                  public_ip_address_dns_name))
+                                                                  public_ip_address_dns_name,
```

Comment:
Per our conversation, we should add a few defaulting logics in the validator. I am combining comments for both vm and network modules.
1. When lb sku is `standard`, create a `standard` public ip with `static` allocation. Optionally (up to you), when an existing public-ip is provided for a new `standard` LB, verify it has the right sku and right allocation.  
2. Apply the same for `az network lb create`
3. For `az network public-ip create`, when `--sku` is standard, default the allocation to `static` or verify user provided value is right.